### PR TITLE
Setting bucket properties is goofy

### PR DESCRIPTION
**Pull request by bkerley**
Developers want to do this:

```ruby
bucket.props['precommit'].push({ 'mod' => 'validate_json', 'fun' => 'validate' })
```

What happens is this fetches the props hash from Riak, and then mutates that has in place. At no time is it saved to Riak.

Doing this should save it:

```ruby
bucket.props['precommit'].push({ :mod => 'validate_json', :fun => 'validate' })
bucket.props = {}
```

But it's terribly unintuitive.

### CODE REVIEW

**Comment by bkerley** [comment]
TODO: test it with composite/hash properties like the modfun above.

File: lib/riak/client/beefcake_protobuffs_backend.rb [hunk 1]
```diff
@@ -13,12 +13,13 @@ def self.configured?
...
           require 'riak/client/beefcake/message_overlay'
           require 'riak/client/beefcake/object_methods'
+          require 'riak/client/beefcake/bucket_properties_operator'
           require 'riak/client/beefcake/crdt_operator'
           require 'riak/client/beefcake/crdt_loader'
           require 'riak/client/beefcake/protocol'
           require 'riak/client/beefcake/socket'
           true
-        rescue LoadError, NameError
+        rescue LoadError, NameError => e
```
Comment (lukebakken) [inline lib/riak/client/beefcake_protobuffs_backend.rb]: Ruby-n00b question: why set `e` when it's unused?

**Comment by lukebakken** [comment]
@bkerley this all looks great and is very easy to follow.

File: lib/riak/client/beefcake_protobuffs_backend.rb (hunk 1 above)
Comment (bkerley) [inline lib/riak/client/beefcake_protobuffs_backend.rb]: I so the exception goes into a variable and I can find it in a debugger. I should probably roll that one back.

Comment (lukebakken) [inline lib/riak/client/beefcake_protobuffs_backend.rb]: Thanks for the explanation.

**Comment by drewkerrigan** [comment]
+1

**Comment by drewkerrigan** [comment]
+1 3257730

**Comment by bkerley** [comment]
@borshop merge

**Pull request merged**